dataset/src/main/scala/frameless/TypedEncoder.scala

```diff
@@ -27,23 +29,26 @@ abstract class TypedEncoder[T](implicit val classTag: ClassTag[T]) extends Seria
 //   - import TypedEncoder.usingInjection
 //   - import TypedEncoder.usingDerivation
 // """)
-object TypedEncoder {
+object TypedEncoder extends TypedEncoder0 {
```

Comment:
If it's not necessary for this PR I would rather see it revered, having ambiguities when defining things you're not supposed to is IMO a good thing :)